homeassistant/components/homekit/type_humidifiers.py

```diff
@@ -0,0 +1,241 @@
+                PROP_MIN_STEP: 1,
+            },
+        )
+
+        # Active/inactive characteristics
+        self.char_active = serv_humidifier_dehumidifier.configure_char(
+            CHAR_ACTIVE, value=False
+        )
+
+        self.async_update_state(state)
+
+        serv_humidifier_dehumidifier.setter_callback = self._set_chars
+
+        self.linked_humidity_sensor = self.config.get(CONF_LINKED_HUMIDITY_SENSOR)
+        if self.linked_humidity_sensor:
+            humidity_state = self.hass.states.get(self.linked_humidity_sensor)
+            if humidity_state:
+                self._async_update_current_humidity(None, None, humidity_state)
+
+    async def run_handler(self):
+        """Handle accessory driver started event.
+
+        Run inside the Home Assistant event loop.
+        """
+        if self.linked_humidity_sensor:
+            async_track_state_change(
+                self.hass,
+                self.linked_humidity_sensor,
+                self._async_update_current_humidity,
+            )
+
+        await super().run_handler()
+
+    @callback
+    def _async_update_current_humidity(
+        self, entity_id=None, old_state=None, new_state=None
+    ):
+        """Handle linked humidity sensor state change to update HomeKit value."""
+        if new_state is None:
+            return
+        try:
+            current_humidity = float(new_state.state)
+            if self.char_current_humidity.value != current_humidity:
+                self.char_current_humidity.set_value(current_humidity)
+                _LOGGER.debug(
+                    "%s: Linked humidity sensor %s changed to %d",
+                    self.entity_id,
+                    self.linked_humidity_sensor,
+                    current_humidity,
+                )
+        except ValueError as ex:
+            _LOGGER.error(
+                "%s: Unable to update from linked humidity sensor %s: %s",
+                self.entity_id,
+                self.linked_humidity_sensor,
+                ex,
+            )
+
+    def _set_chars(self, char_values):
+        _LOGGER.debug("HumidifierDehumidifier _set_chars: %s", char_values)
+
+        if CHAR_ACTIVE in char_values:
+            self.call_service(
+                DOMAIN,
+                SERVICE_TURN_ON if char_values[CHAR_ACTIVE] else SERVICE_TURN_OFF,
+                {ATTR_ENTITY_ID: self.entity_id},
+                f"{CHAR_ACTIVE} to {char_values[CHAR_ACTIVE]}",
+            )
+        if self._target_humidity_char_name in char_values:
+            humidity = round(char_values[self._target_humidity_char_name])
+            self.call_service(
+                DOMAIN,
+                SERVICE_SET_HUMIDITY,
+                {ATTR_ENTITY_ID: self.entity_id, ATTR_HUMIDITY: humidity},
+                f"{self._target_humidity_char_name} to {char_values[self._target_humidity_char_name]}{UNIT_PERCENTAGE}",
+            )
+        if CHAR_TARGET_HUMIDIFIER_DEHUMIDIFIER in char_values:
```

Comment:
It looks like its possible to 

call
`SERVICE_TURN_ON`
`SERVICE_SET_HUMIDITY`
`SERVICE_TURN_OFF`

In a single call.  We should avoid turning on and off in the same call

Reply:
Fixed.